pkg/tree/tree.go

```diff
@@ -134,7 +134,7 @@ func (t Tree) Prune() {
 	}
 }
 
-// Empty returns true iff the Tree contains no leaf values.
```

Comment:
I'm not sure this is a typo. I think the intent with "iff" is to say "if and only if". I think the distinction here is logical. When the tree is empty, it is necessary that the tree contains no leaf values. But without the `iff` distinction, it is not necessary for no leaf values to guarantee an empty tree (there could be other requirements for a tree to be empty, not just zero leaf values).

That's why we have this equivalence here, we want the statement to be bidirectional:
* If a tree is empty, it has no leaf values.
* If a tree has no leaf values, it is empty.

a tree is empty iff it contains no leaf values.

At least, this is how I read it. I hope it makes sense; for the purpose of clarity, I'd leave this as is :)

Reply:
Thank you very much @mateiidavid. Yes, it makes sense to me now.